gobblin-utility/src/main/java/gobblin/util/HadoopUtils.java

```diff
@@ -152,8 +153,31 @@ public static void movePath(FileSystem srcFs, Path src, FileSystem dstFs, Path d
     if (srcFs.getUri().equals(dstFs.getUri())) {
       renamePath(srcFs, src, dst);
     } else {
-      if (!FileUtil.copy(srcFs, src, dstFs, dst, true, false, dstFs.getConf())) {
-        throw new IOException(String.format("Failed to move %s to %s", src, dst));
+      // Determining if the source filesystem in local will allow us to call moveFromLocalFile and give the
+      // destination filesystem the ability optimize the transfer. For example, when the source file is
+      // local the S3AFileSystem will skip buffering it into the fs.s3a.buffer.dir and immediately transfer the
+      // file to S3.
+      boolean isSourceFileSystemLocal = srcFs instanceof LocalFileSystem;
```

Comment:
Why do you need this optimization exactly? The method `FileSystem.moveFromLocalFile` just invokes `FileUtil.copy` anyway, to its really the same thing.